spec/syncSpec.js

```diff
@@ -641,11 +663,13 @@ function stopSync(dataset) {
  *
  * @param {string} expectedEvent - The name of the event to wait for.
  */
-function waitForSyncEvent(expectedEvent) {
+function waitForSyncEvent(expectedEvent, validator) {
   return function() {
+    const eventValidator = validator || noop;
+
     return new Promise(function(resolve) {
       $fh.sync.notify(function(event) {
-        if (event.code === expectedEvent) {
+        if (event.code === expectedEvent && eventValidator(event)) {
```

Comment:
Just wondering about this eventValidator. Most test use the resolved event and then have a separate then clause to validate the event. Is the use case different here?